File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // map<int,int>mp;
        // for(int i=0;i<nums.size();i++){
        //     int a=nums[i];
        //     int b=target-a;
        //     if(mp.find(b)!=mp.end()){
        //         return{i,mp[b]};
        //     }
        //     mp[a]=i;
        // }
        // return {};

        vector<pair<int,int>>v;
        int n=nums.size();
        for(int i=0;i<n;i++){
            v.push_back({nums[i],i});
        }
        sort(v.begin(),v.end());
        int lo=0,hi=n-1;
        while(lo<hi){
            int sum=v[lo].first+v[hi].first;
            if(sum==target){
                return {v[lo].second,v[hi].second};
            }else if(sum>target){
                hi--;
            }else{
                lo++;
            }
        }
        return{-1,-1};
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # n=len(nums)
        # for i in range(n):
        #     for j in range(i+1,n):
        #         if nums[i]+nums[j]==target:
        #             return [i,j]
        # return [-1,-1]

        n=len(nums)
        v=[]
        for i in range(n):
            v.append((nums[i],i))
        v.sort()
        low=0
        high=n-1
        while low<high:
            sum=v[low][0]+v[high][0]
            if target==sum:
                return [v[low][1],v[high][1]]
            elif sum<target:
                low+=1
            else:
                high-=1
        return [-1,-1]